lnmock/preimage_cache.go

```diff
@@ -0,0 +1,45 @@
+package lnmock
```

Comment:
I am not sure about a mock package. Whether it will again introduce circular dep issues. In general I'd think it is best to define the mock in the same package as the real implementation, so that there can't be new circular dependencies.

Reply:
all of the circular dependency issues wrt to mocks are solved by writing unit tests in `package name_test`. there can't be circular dependencies because `name_test` can't be imported by anything else, and the mocks aren't to be used in the real implementation.

if we add them in the same package but make them private, they'll be inaccessible from the tests that do so. if we make them public to access from tests, then we have to document them to satisfy the linter but they also show up on godocs.

Reply:
Ok. Yes, in another project I actually do the same. Created a separate package containing test code to be able to reuse it without exposing it all from the real implementation.

Reply:
I'm not sure about this either. Maybe the `preimage_cache` should have its own package, which can contain both the real, and the mock implementation?

Either way, I think this discussion can be made its own PR, as the actual consistency fix is what is important here.